### PR TITLE
ezFoundation improvements

### PR DESCRIPTION
**Pull request by jankrassnigg**

### CODE REVIEW

File: Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h [hunk 1]
```diff
@@ -3,11 +3,12 @@
 #include <Foundation/Containers/HybridArray.h>
 
 /// \brief This allocation policy works like a stack. You can only "push" and "pop" allocations
-///   in the correct order.
+///   in the correct order. The allocation policy can poison all allocations to show usages after destruction.
```
Comment (C-Core) [inline Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h]: "poison" sounds dangerous although it is supposed to help the user. Maybe something like "OverwriteMemoryOnReset" would be more descriptive?

File: Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h [hunk 2]
```diff
@@ -3,11 +3,12 @@
 #include <Foundation/Containers/HybridArray.h>
 
 /// \brief This allocation policy works like a stack. You can only "push" and "pop" allocations
```
Comment (C-Core) [inline Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h]: Btw this is not a stack allocation policy but a linear allocation policy. The name is misleading and the description is just wrong.

File: Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h (hunk 1 above)
Comment (jankrassnigg) [reply Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h]: Fixed

File: Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h (hunk 2 above)
Comment (jankrassnigg) [reply Code/Engine/Foundation/Memory/Policies/AllocPolicyStack.h]: Renamed it and fixed the description.

File: Code/Engine/Foundation/Memory/FrameAllocator.h [hunk 3]
```diff
@@ -6,7 +6,12 @@
 class EZ_FOUNDATION_DLL ezDoubleBufferedLinearAllocator
 {
 public:
-  using StackAllocatorType = ezLinearAllocator<ezAllocatorTrackingMode::Basics>;
+#if EZ_ENABLED(EZ_COMPILE_FOR_DEBUG)
+  static constexpr bool poisonMemory = true;
```
Comment (C-Core) [inline Code/Engine/Foundation/Memory/FrameAllocator.h]: more "poison"... ;)

Comment (jankrassnigg) [reply Code/Engine/Foundation/Memory/FrameAllocator.h]: yeah, also noticed just now, all fixed now

**Review by C-Core — APPROVED** [review]

**Pull request merged**